src/web/views/Content/Select/index.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import { Modal, Button, Radio, OverlayTrigger, Tooltip, Panel, Well } from 'react-bootstrap'
+import Select from 'react-select'
+import axios from 'axios'
+import classnames from 'classnames'
+
+import CreateModal from '../modal'
+import {
+  fetchContentMessagesRecent,
+  fetchContentMessagesCount,
+  fetchContentCategories,
+  upsertContentMessages
+} from '~/actions'
+
+const style = require('./style.scss')
+
+class SelectContent extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { show: false, category: null, searchTerm: '', contentToEdit: null }
+
+    window.botpress = window.botpress || {}
+    window.botpress.pickContent = (options = {}, callback) => {
+      this.props.fetchContentMessagesRecent({})
+      this.props.fetchContentMessagesCount()
+      this.props.fetchContentCategories()
+      this.setState({ show: true, callback })
+    }
+  }
+
+  componentWillUnmount() {
+    delete window.botpress.pickContent
+  }
+
+  search = event => {
+    this.setState({ searchTerm: event.target.value })
+    this.props.fetchContentMessagesRecent({ searchTerm: event.target.value })
+  }
+
+  handleCreate = () => {
+    this.props
+      .upsertContentMessages({ categoryId: this.state.category.id, formData: this.state.contentToEdit })
+      .then(() =>
+        Promise.all([
+          this.props.fetchContentMessagesRecent({ searchTerm: this.state.searchTerm }),
+          this.props.fetchContentMessagesCount()
+        ])
+      )
+      .then(() => this.setState({ category: null, contentToEdit: null }))
+  }
+
+  handlePick(item) {
+    this.setState({ show: false })
+    this.state.callback(item)
+  }
+
+  handleFormEdited = data => {
+    this.setState({ contentToEdit: data })
+  }
+
+  render() {
+    const props = this.props
+    const noop = () => {}
+
+    const onClose = () => this.setState({ show: false })
```

Comment:
both of these can be instance props instead, right?

Reply:
Done! )